Util_Environment.php

```diff
@@ -1677,39 +1677,31 @@ public static function array_intersect_partial( array $array1, array $array2 ):
 	 *
 	 * @return int
 	 */
-	public static function get_cron_schedule_time( int $cron_time = 0 ): bool {
-		// WordPress timezone.
-		$timezone_string = get_option( 'timezone_string' );
-		$wp_timezone     = new \DateTimeZone( $timezone_string ? $timezone_string : 'UTC' );
-		$server_timezone = new \DateTimeZone( date_default_timezone_get() );
+	public static function get_cron_schedule_time(int $cron_time = 0): int {
+		// Fetch the current user's timezone from user meta.
+		$user_timezone = new \DateTimeZone( get_user_meta( get_current_user_id(), 'timezone', true ) ?: wp_timezone()->getName() );
```

Comment:
Users don't have an option to choose a timezone different from the WP site timezone, so there is no "timezone" in user meta.  You can just set the variable to `wp_timezone()`.  If it's only used once below, just use `wp_timezone()` there.

Reply:
Updated